libtelemetry/src/main/java/com/mapbox/android/telemetry/VisionObjectDetectionEvent.java

```diff
@@ -0,0 +1,212 @@
+    this.clazz = null;
+    this.sign_value = null;
+    this.object_size_height = null;
+    this.object_size_width = null;
+    this.object_pos_height = null;
+    this.distance_from_camera = null;
+  }
+
+  public String getEvent() {
+    return event;
+  }
+
+  public String getCreated() {
+    return created;
+  }
+
+  public double getObjectLatitude() {
+    return object_lat;
+  }
+
+  public void setObjectLatitude(double latitude) {
+    this.object_lat = latitude;
+  }
+
+  public double getObjectLongitude() {
+    return object_lon;
+  }
+
+  public void setObjectLongitude(double longitude) {
+    this.object_lon = longitude;
+  }
+
+  public double getVehicleLatitude() {
+    return vehicle_lat;
+  }
+
+  public void getVehicleLatitude(double latitude) {
+    this.vehicle_lat = latitude;
+  }
+
+  public double getVehicleLongitude() {
+    return vehicle_lon;
+  }
+
+  public void setVehicleLongitude(double longitude) {
+    this.vehicle_lon = longitude;
+  }
+
+  public String getClazz() {
+    return clazz;
+  }
+
+  public void setClazz(String clazz) {
+    this.clazz = clazz;
+  }
+
+  public String getSignValue() {
+    return sign_value;
+  }
+
+  public void setSignValue(String sign_value) {
+    this.sign_value = sign_value;
+  }
+
+  public double getObjectSizeWidth() {
+    return object_size_width;
+  }
+
+  public void setObjectSizeWidth(double object_size_width) {
+    this.object_size_width = object_size_width;
+  }
+
+  public double getObjectSizeHeight() {
+    return object_size_height;
+  }
+
+  public void setObjectSizeHeight(double object_size_height) {
+    this.object_size_height = object_size_height;
+  }
+
+  public double getObjectPositionHeight() {
+    return object_pos_height;
+  }
+
+  public void setObjectPositionHeight(double object_pos_height) {
+    this.object_pos_height = object_pos_height;
+  }
+
+  public double getDistanceFromCamera() {
+    return distance_from_camera;
+  }
+
+  public void setDistanceFromCamera(double distance_from_camera) {
+    this.distance_from_camera = distance_from_camera;
+  }
+
+  @Override
+  Event.Type obtainType() {
+    return Event.Type.VIS_GENERAL;
+  }
+
+  private VisionObjectDetectionEvent(Parcel in) {
+    this.event = in.readString();
+    this.created = in.readString();
+    this.object_lat = in.readDouble();
+    this.object_lon = in.readDouble();
+    this.vehicle_lat = in.readDouble();
+    this.vehicle_lon = in.readDouble();
+    this.clazz = in.readString();
+    this.sign_value = in.readString();
+    this.object_size_width = in.readDouble();
+    this.object_size_height = in.readDouble();
+    this.object_pos_height = in.readDouble();
+    this.distance_from_camera = in.readDouble();
+  }
+
+  @Override
+  public int describeContents() {
+    return 0;
+  }
+
+  @Override
+  public void writeToParcel(Parcel dest, int flags) {
+    dest.writeString(event);
+    dest.writeString(created);
+
+    writeDoubleIfNotNull(dest, object_lat);
```

Comment:
Will it work with reading then after writing fields optionally?

Reply:
you're right. it will make more sense to make use of binary flags - my bad. fixing it!

Reply:
Or just writing default values - if that will be easier

Reply:
IMHO flag is a bit cleaner. it's hard sometimes to select default/invalid values for params like lat/lon.